diagnostics-app/src/app/diagnostics/diagnostics-card/diagnostics-card.component.html

```diff
@@ -0,0 +1,55 @@
+<!-- 
+ Copyright 2023 The Chromium Authors. All rights reserved.
+ Use of this source code is governed by a BSD-style license that can be
+ found in the LICENSE file.
+-->
+
+<div class="diagnostics-card">
+  <div class="header-bar">
+    <div class="header-and-params">
+      <header>{{ routine.toUpperCase() }}</header>
+      <div class="params-info" *ngIf="params">
+        <div class="param-info" *ngFor="let param of $any(params) | keyvalue: originalOrder">
+          {{ param.key }}: {{ param.value }}
+        </div>
+      </div>
+    </div>
+    <div class="buttons">
+      <button mat-raised-button color="basic" (click)="startRoutine()" [disabled]="state !== 0">Run</button>
+      <button mat-raised-button color="warn" (click)="stopRoutine()" [disabled]="state !== 1">Stop</button>
+      <button mat-raised-button color="basic" (click)="resumeRoutine()" [disabled]="!needResume() || state !== 2">Resume</button>
+    </div>
+  </div>
+  <ng-container *ngIf="intervalId">
+    <ng-container *ngIf="error; then showError; else showRunning"></ng-container>
+    <ng-template #showError>
+      <div class="error-message">
+        <mat-icon mat-list-icon>warning</mat-icon>
+        {{ error }}
+      </div>
+    </ng-template>
+    <ng-template #showRunning>
+      <div class="progress-bar-container">
+        <mat-progress-bar class="progress-bar" mode="determinate" [value]="routineInfo?.progress_percent"></mat-progress-bar>
+        {{ routineInfo?.progress_percent }}%
+      </div>
+    </ng-template>
+  </ng-container> 
+  <div class="message-bar" *ngIf="routineInfo">
+    <div class="passed" *ngIf="routineInfo.status === 'passed'">Passed</div>
```

Comment:
Can we use `NgClass` to conditionally change the color here?

Reply:
Sure. Done.